design/open/spark-outputcommitter/committer.md

```diff
@@ -187,8 +211,17 @@ the same paths but keeping things safe regardless.
 [^2]: 	Whenever there is a conflict, we want the task branch (which will become
     the "latest writer" after a successful HOC commit) to win.
 
+# Potential wins
+
+* Explicit requests _not_ to write 3 times and give better behaviour than
+  the default FileOutputCommitter from Spark have appeared on our Slack
+  [#data-architecture-discussion][slack-dont-write-thrice].
+* Multiple users have requested "overwrite" save mode.
+* Multiple users have requested multi-writer support.
```

Comment:
I would specify how the LakeFSOutputCommitter would support that (basically mentioning that some writes would fail if multiple writes concurrently occur?)

Reply:
Actually that would depend on [SaveMode](https://spark.apache.org/docs/3.2.1/api/java/index.html?org/apache/spark/sql/SaveMode.html) I think:

* For append, everything is sort-of concatenated (I'm not sure that Spark actually does this, specifically as far as multiple values for the same key are concerned).
* For ErrorIfExists, all but one ("the first") writer should error out -- even in the face of concurrent writes.
* For ignore, the write should silently do nothing.
* For overwrite, "latest" writer wins.

I'd prefer not to do the research right now :-/